### PR TITLE
rpcserver: forward gRPC proxy requests to localhost when listening on all interfaces

### LINKED ISSUE #1567: [Bug] LND REST Proxy stop working if gRPC is exposed on 0.0.0.0
### Background

When I added `rpclisten=0.0.0.0:10009`, then the rest endpoint sends me error 500:

Config:
```
restlisten=0.0.0.0:8080
rpclisten=0.0.0.0:10009
bitcoin.node=bitcoind
bitcoind.rpchost=bitcoind:43782
bitcoind.zmqpath=tcp://bitcoind:28332
externalip=testlnd1.btcpayserver.com:9735
alias=
noencryptwallet=1
notls=1
bitcoin.active=1
bitcoin.mainnet=1
```

Error:
```
{"error":"connection error: desc = \"transport: authentication handshake failed: x509: certificate is valid for 127.0.0.1, ::1, 172.18.0.6, not 0.0.0.0\"","code":13}
```

# Workaround

Adding an unused rpclisten endpoint fix the situation.
```
restlisten=0.0.0.0:8080
rpclisten=127.0.0.1:10008
rpclisten=0.0.0.0:10009
bitcoin.node=bitcoind
bitcoind.rpchost=bitcoind:43782
bitcoind.zmqpath=tcp://bitcoind:28332
externalip=testlnd1.btcpayserver.com:9735
alias=
noencryptwallet=1
notls=1
bitcoin.active=1
bitcoin.mainnet=1
```

### PR DESCRIPTION
**Pull request by wpaulino**
This prevents certificate issues when accessing the gRPC REST proxy externally.

Fixes #1567.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Oh I get it! thanks a lot, will test.

**Comment by jamaljsr** [comment]
I also ran into the "authentication handshake failed" error on my remote node, trying to setup the joule extension. I manually patched `lnd` with these changes, rebuilt and now the REST api works as expected. Thanks @wpaulino 👍

File: rpcserver.go [hunk 1]
```diff
@@ -533,9 +533,17 @@ func (r *rpcServer) Start() error {
 	// TODO(roasbeef): eventually also allow the sub-servers to themselves
 	// have a REST proxy.
 	mux := proxy.NewServeMux()
+	grpcEndpoint := cfg.RPCListeners[0].String()
```
Comment (halseth) [inline rpcserver.go]: nit: add a comment why this string replacing is needed.

Comment (wpaulino) [reply rpcserver.go]: Done.

**Pull request merged**